hicommonwealth/commonwealth
# Delete Substrate Proposal Viewing

## Link to Issue
Closes: #4699
Closes: #5284

## Description of Changes
- Deletes all Substrate proposal pages, types, and controllers.
- Migration which removes Substrate chain-event subscriptions, notifications, and refactors proposal links into web links.

## "How We Fixed It"

- Fixed #5284 by updating the `showSidebar` component property in `ViewThreadPage.tsx`.
- Fixed #5285 by removing legacy Substrate identity user rendering in `user.tsx`

## Test Plan
- Login with Polkadot wallet
- Join a Substrate community i.e. 'edgeware'
- View and interact with a Substrate community by creating a thread, commenting, and reacting.
- View and click on the new web links by finding a post that previously had proposals linked (try ordering threads by like -> there should be a 'Pizza Party' thread that had proposals and now has web links).
- Now create your own thread and use web links to link any website.
- Ensure proposals/referenda/tips aren't options for Substrate communities in the left-hand sidebar.
- View Aave and Osmosis proposal pages to ensure they still function correctly.

## Deployment Plan

**WARNING: MUST DO STEP (1) BEFORE DEPLOYMENT SINCE ENV VAR IS USED IN MIGRATION**
1. `Set CE_DATABASE_URL` env var to the chain-events database URL
3. Deploy to CW
4. Delete the env var once the migration is complete since the database URL is subject to change over time.

## Other Considerations

- `Set CE_DATABASE_URL` needs to be set in the QA Heroku env in order for the migration to complete successfully. Failing to set the env var will _not_ cause the migration to fail but Substrate proposal links will not be updated and therefore cannot be tested.
